File: TheAnvil/src/main/java/com/qualityplus/anvil/listener/VanillaAnvilListener.java
package com.qualityplus.anvil.listener;

import com.qualityplus.anvil.api.box.Box;
import com.qualityplus.anvil.base.gui.anvilmain.AnvilMainGUI;
import com.qualityplus.anvil.base.session.AnvilSessionImpl;
import com.qualityplus.assistant.lib.com.cryptomorin.xseries.XMaterial;
import com.qualityplus.assistant.lib.eu.okaeri.injector.annotation.Inject;
import com.qualityplus.assistant.lib.eu.okaeri.platform.core.annotation.Component;
import com.qualityplus.assistant.util.block.BlockUtils;
import org.bukkit.block.Block;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.player.PlayerInteractEvent;

import java.util.stream.Collectors;
import java.util.stream.Stream;

@Component
public final class VanillaAnvilListener implements Listener {
    private @Inject Box box;

    @EventHandler
    public void onOpenBrewingStand(PlayerInteractEvent e) {
        if (e.getAction() != Action.RIGHT_CLICK_BLOCK) return;

        Block block = e.getClickedBlock();

        if (BlockUtils.isNull(block)) return;

        if (!blockIsAnvil(block)) return;

        if (!box.files().config().openAsVanillaAnvil) return;

        e.setCancelled(true);

        e.getPlayer().openInventory(new AnvilMainGUI(box, new AnvilSessionImpl(null, null, null)).getInventory());
    }

    private boolean blockIsAnvil(Block block) {
        return Stream.of(XMaterial.ANVIL, XMaterial.CHIPPED_ANVIL, XMaterial.DAMAGED_ANVIL)
                .map(XMaterial::parseMaterial)
                .collect(Collectors.toList())
                .contains(block.getType());
    }
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/listener/VanillaAnvilListener.kt
package com.qualityplus.anvil.listener

import com.cryptomorin.xseries.XMaterial
import com.qualityplus.anvil.api.box.Box
import com.qualityplus.anvil.base.gui.anvilmain.AnvilMainGUI
import com.qualityplus.anvil.base.session.AnvilSessionImpl
import com.qualityplus.assistant.util.block.BlockUtils
import eu.okaeri.injector.annotation.Inject
import eu.okaeri.platform.core.annotation.Component
import org.bukkit.block.Block
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.Action
import org.bukkit.event.player.PlayerInteractEvent
import java.util.stream.Collectors
import java.util.stream.Stream

@Component
class VanillaAnvilListener : Listener {
    @Inject
    private lateinit var box: Box

    @EventHandler
    fun onOpenBrewingStand(e: PlayerInteractEvent) {
        if (e.action != Action.RIGHT_CLICK_BLOCK) {
            return
        }

        val block = e.clickedBlock

        if (BlockUtils.isNull(block)) {
            return
        }

        if (!blockIsAnvil(block)) {
            return
        }

        if (!box.getFiles()!!.config()!!.openAsVanillaAnvil) {
            return
        }

        e.isCancelled = true

        e.player.openInventory(AnvilMainGUI(box, AnvilSessionImpl(null, null, null)).inventory)
    }

    private fun blockIsAnvil(block: Block?): Boolean {
        return Stream.of(XMaterial.ANVIL, XMaterial.CHIPPED_ANVIL, XMaterial.DAMAGED_ANVIL)
            .map { obj: XMaterial -> obj.parseMaterial() }
            .collect(Collectors.toList())
            .contains(block!!.type)
    }
}